### PR TITLE
Add tag related endpoints to drip-mock.

### PR DESCRIPTION
**Pull request by JuhaSydanmaa**
Also fixed subscribers/<subscriber_id> endpoint

https://github.com/VRTFinland/VRTfront/issues/3409

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -38,11 +40,20 @@ def hello_world(account_id):
     return "Woot woot"
 
 
-@app.route("/v2/<account_id>/subscribers/<subscriber_id>/", methods=["GET"])
+@app.route("/v2/<account_id>/subscribers/<subscriber_id>", methods=["GET"])
 def subscriber_handler(account_id, subscriber_id):
-    if subscriber_id in dripData.subscribers:
-        return dripData.subscribers[subscriber_id]
-
+    subscriber = dripData.subscribers[subscriber_id]
```
Comment (pingviini) [inline app.py]:
```suggestion
    subscriber = dripData.subscribers.get(subscriber_id, None)
```

File: app.py [hunk 2]
```diff
@@ -87,6 +98,35 @@ def events(account_id):
     dripData.events.append(request.json["events"][0])
     return Response(None, status=204)
 
+@app.route("/v2/<account_id>/tags", methods=["GET", "POST"])
```
Comment (pingviini) [inline app.py:106]: In Python coding conventions functions should be separated by 2 empty lines.

```suggestion

@app.route("/v2/<account_id>/tags", methods=["GET", "POST"])
```

File: app.py [hunk 3]
```diff
@@ -87,6 +98,35 @@ def events(account_id):
...
+def tags(account_id):
+    if request.method == "GET":
+        return {
+            "tags" : dripData.tags
+        }
+    elif request.method == "POST":
+        data = request.json.get("tags", [])[0]
```
Comment (pingviini) [inline app.py]: This will throw IndexError if there aren't any tags.

File: app.py [hunk 4]
```diff
@@ -87,6 +98,35 @@ def events(account_id):
...
+        if subscriber_id:
+            subscriber = dripData.subscribers[subscriber_id]
+            subscriber_tags = subscriber["tags"]
+            if new_tag not in subscriber_tags:
+                subscriber_tags.append(new_tag)
+            if new_tag not in dripData.tags:
+                dripData.tags.append(new_tag)
+        return Response(response={}, status=201)
+
+@app.route("/v2/<account_id>/subscribers/<email>/tags/<tag_name>", methods=["DELETE"])
```
Comment (pingviini) [inline app.py:121]:
```suggestion

@app.route("/v2/<account_id>/subscribers/<email>/tags/<tag_name>", methods=["DELETE"])
```

File: app.py [hunk 5]
```diff
@@ -87,6 +98,35 @@ def events(account_id):
...
+def delete_tag(account_id, email, tag_name):
+    subscriber_id = dripData.email_id_map.get(email)
+    if subscriber_id:
+        subscriber = dripData.subscribers[subscriber_id]
+        subscriber_tags = subscriber["tags"]
+        if tag_name in subscriber_tags:
+            subscriber_tags.remove(tag_name)
+    return Response(None, status=204)
```
Comment (pingviini) [inline app.py:129]:
```suggestion
    return Response(None, status=204)
    
```

File: app.py [hunk 6]
```diff
@@ -87,6 +98,35 @@ def events(account_id):
...
+            if new_tag not in dripData.tags:
+                dripData.tags.append(new_tag)
+        return Response(response={}, status=201)
+
+@app.route("/v2/<account_id>/subscribers/<email>/tags/<tag_name>", methods=["DELETE"])
+def delete_tag(account_id, email, tag_name):
+    subscriber_id = dripData.email_id_map.get(email)
+    if subscriber_id:
+        subscriber = dripData.subscribers[subscriber_id]
+        subscriber_tags = subscriber["tags"]
```
Comment (pingviini) [inline app.py:126]:
```suggestion
        subscriber_tags = subscriber.get("tags", [])
```

File: app.py (hunk 5 above)
Comment (pingviini) [inline app.py]:
```suggestion

@app.route("/v2/<account_id>/subscribers/<email>/tags/<tag_name>", methods=["DELETE"])
def delete_tag(account_id, email, tag_name):
    subscriber_id = dripData.email_id_map.get(email)
    if subscriber_id:
        subscriber = dripData.subscribers[subscriber_id]
        subscriber_tags = subscriber.get("tags", [])
        if tag_name in subscriber_tags:
            subscriber_tags.remove(tag_name)
            return Response(None, status=204)

    return Response(None, status=404)

```

Comment (JuhaSydanmaa) [reply app.py]: Changed

File: app.py (hunk 3 above)
Comment (JuhaSydanmaa) [reply app.py]: Fixed

File: app.py (hunk 2 above)
Comment (JuhaSydanmaa) [reply app.py:106]: Added 2 empty lines between functions

File: app.py (hunk 1 above)
Comment (JuhaSydanmaa) [reply app.py]: Changed

File: app.py [hunk 7]
```diff
@@ -88,6 +103,41 @@ def events(account_id):
...
+            subscriber_id = dripData.email_id_map.get(email)
+            if subscriber_id:
+                subscriber = dripData.subscribers[subscriber_id]
+                subscriber_tags = subscriber["tags"]
+                if new_tag not in subscriber_tags:
+                    subscriber_tags.append(new_tag)
+                if new_tag not in dripData.tags:
+                    dripData.tags.append(new_tag)
+            return Response(response={}, status=201)
+        return Response(None, status=402)
```
Comment (pingviini) [inline app.py]: 402 is a bit strange status code in here. Maybe 400 would be correct in here?

Comment (pingviini) [inline app.py]:
```suggestion
        return Response(None, status=400)
```

Comment (JuhaSydanmaa) [reply app.py]: That's a brainfart, it was supposed to be 400.

Comment (JuhaSydanmaa) [reply app.py]: Changed

**Review by pingviini — APPROVED** [review]

**Pull request merged**